File: Difficulty: Easy/Second Largest/second-largest.py
class Solution:
    def getSecondLargest(self, arr):
        n = len(arr)
        if n < 2:
            return -1
        large = float('-inf')
        s_large = float('-inf')
        
        for num in arr:
            if num > large:
                s_large = large
                large = num
            elif num > s_large and num != large:
                s_large = num
        
        return s_large if s_large != float('-inf') else -1


#{ 
 # Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        arr = list(map(int, input().split()))
        ob = Solution()
        ans = ob.getSecondLargest(arr)
        print(ans)
        print("~")
# } Driver Code Ends

File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        int sec_max = -1;
        int max_ele = INT_MIN;
        
        for(int i = 0; i < arr.size(); i++){
            if(arr[i] > max_ele) max_ele = arr[i];
        }
        
        for(int j = 0; j < arr.size(); j++){
            if(arr[j] != max_ele){
                sec_max = max(sec_max, arr[j]);
            }
        }
        
        return sec_max;
    }
};